### DISCUSSION TITLE
autokeras works for imbalanced data?

### DISCUSSION THREAD

**Discussion by sasi29** [keras-team/autokeras#1357] [General]
How does autokeras work for imbalanced data which is in a structured format? there is no option for weights

**Comment by haifeng-jin** [comment]
Is the class_weight [here](https://keras.io/api/models/model_training_apis/#fit-method) that you are looking for?
AutoKeras support all these keras params in the fit function, too.

**Reply by brunosdiniz** [reply]
What is the expected format to pass this dict when we have multiple ClassificationHead outputs with different classes using AutoModel?
I tried passing a list to fit method like below but got an error:

**Model definition:**
```
model = ak.AutoModel( 
      inputs     = [ak.TextInput(), ak.StructuredDataInput()]
    , outputs    = [ ak.ClassificationHead(loss='categorical_crossentropy', metrics=['accuracy'])
                   , ak.ClassificationHead(loss='categorical_crossentropy', metrics=['accuracy'])
                   ]
    , overwrite  = True
    , max_trials = 10
)
```

**Dictionaries:**
```
{1: 82.33333333333333, 85: 329.3333333333333, 86: 2.6775067750677506, 123: 0.22838650023116042, 143: 23.523809523809526, 185: 329.3333333333333, 329: 82.33333333333333, 344: 329.3333333333333, 377: 0.14578722148443263, 632: 329.3333333333333, 634: 3.3265993265993266, 642: 109.77777777777777}
{1: 0.12455874936964195, 3: 9.686274509803921, 4: 8.032520325203253, 5: 21.955555555555556, 7: 0.37985390234525185, 9: 65.86666666666666, 11: 47.04761904761905, 12: 3.4305555555555554, 13: 109.77777777777777, 81: 4.63849765258216, 82: 1.9603174603174602, 103: 329.3333333333333}
```

**Attempt to pass a list of dicts:**
```
model.fit(...
, class_weight = [dict1, dict2]
)
```

**Error:**
```
~/.local/lib/python3.7/site-packages/tensorflow/python/keras/engine/data_adapter.py in _make_class_weight_map_fn(class_weight)
   1379     weighting.
   1380   """
-> 1381   class_ids = list(sorted(class_weight.keys()))
   1382   expected_class_ids = list(range(len(class_ids)))
   1383   if class_ids != expected_class_ids:

AttributeError: 'list' object has no attribute 'keys'
```

From the looks of it the code might not be expecting a list of dicts in this parameter, but a single dict instead, but there are cases where there might be overlapping classes when using it for multiple outputs.

Any help is much appreciated!

**Comment by brunosdiniz** [comment]
What is the expected format to pass this dict when we have multiple ClassificationHead outputs with different classes using AutoModel?
I´d appreciate if someone could take a look at [my previous comment](https://github.com/keras-team/autokeras/discussions/1357#discussioncomment-1781843) in this discussion.
